### PR TITLE
Doc: Fix BSD docs regarding gmake

### LINKED ISSUE #14404: ./configure error on FreeBSD
Error when running `./configure` on FreeBSD:

> config.status: executing depfiles commands
> config.status: error: in \`/home/murray/tmp/bitcoin\':
> config.status: error: Something went wrong bootstrapping makefile fragments
>     for automatic dependency tracking.  Try re-running configure with the
>     '--disable-dependency-tracking' option to at least be able to build
>     the package (albeit without support for automatic dependency tracking).
> See `config.log' for more details

Relevant config.log:

> config.status:1696: executing depfiles commands
> config.status:1771: cd src       && sed -e '/# am--include-marker/d' Makefile         | make -f - am--depfiles
> make: exec(../build-aux/install-sh) failed (No such file or directory)
> *** Error code 1
> 
> Stop.
> make: stopped in /usr/home/murray/tmp/bitcoin/src
> config.status:1778: $? = 1
> config.status:1782: error: in `/home/murray/tmp/bitcoin':
> config.status:1788: error: Something went wrong bootstrapping makefile fragments
>     for automatic dependency tracking.  Try re-running configure with the
>     '--disable-dependency-tracking' option to at least be able to build
>     the package (albeit without support for automatic dependency tracking).

FreeBSD 11.2, but the issue also existed on 11.1

### PR DESCRIPTION
**Pull request by abitfan**
Fixes #14404

### CODE REVIEW

**Review by murrayn — CHANGES_REQUESTED** [review]
Adding `MAKE=gmake` as an argument to the `./configure` command line examples might be preferable, as `setenv` is specific to `csh`.

**Comment by laanwj** [comment]
I'm still confused why this is needed. I've been running a node on a FreeBSD machine for years, and never had to redefine `MAKE`.

Do you need this when building other autoconf projects as well? If not, I wonder what is different in our build system.

**Comment by abitfan** [comment]
@murrayn There are at least two ./configure examples in each doc file and more can be added later on, imho it just clutters the documentation but it can be changed if deemed necessary. On top of that csh is the default shell on BSDs and compiling bitcoin from source requires some knowledge of the OS anyway.

@laanwj You ran ./configure with --disable-dependency-tracking which won't trigger the problem. This behavior is reproducible on any fresh bsd install. 
"Do you need this when building other autoconf projects...". Yes if you use **gmake** since autoconf will by default use the system's make.

**Comment by laanwj** [comment]
So it's an alternative for `--disable-dependency-tracking`? Makes sense then.

File: doc/build-freebsd.md [hunk 1]
```diff
@@ -36,7 +36,10 @@ export BDB_PREFIX="$PWD/db4"
 
 ## Building Bitcoin Core
 
-**Important**: Use `gmake` (the non-GNU `make` will exit with an error):
+**Important**: Use `gmake` (the non-GNU `make` will exit with an error).
+```shell
+setenv MAKE gmake
+```
```
Comment (vasild) [inline doc/build-freebsd.md]: `setenv` is not portable between shells. Yes, it works on the default `csh` but does not work in others.

Given that `./configure MAKE=gmake` is an equivalent and works with any shell, I would suggest to add `MAKE=gmake` a few lines below where we show how to execute `./configure`.

**Comment by fanquake** [comment]
I agree with the other commenters in using [`./configure MAKE=gmake`](https://github.com/bitcoin/bitcoin/issues/14404#issuecomment-431565631).

**Comment by fanquake** [comment]
Picked this up in #18340.

**Pull request closed without merging**